File: Common/Models/AgentModel.cs
using Common.Models.Base;
using Domain.Enum;

namespace Common.Models;

public class AgentModel : BaseModel
{
    public string Name { get; set; }

    public LegalAgentType TypeId { get; set; }
    
    public string Type { get; set; }

    public long FiscalCode { get; set; }

    public long NumberOfAccounts { get; set; }
}

File: ui/src/entities/agent/model/types/AgentModel.ts
import {ModelConcurrencyTokenInterface} from "../../../types";
import {LegalAgentType} from "../../../../shared/lib/enum";

export interface AgentModel extends ModelConcurrencyTokenInterface {
    id: number,
    name: string,
    typeId: LegalAgentType,
    type: string,
    fiscalCode: number,
    numberOfAccounts: number
}